### PR TITLE
convenience fluent methods for sym/nd

### PR DESCRIPTION
**Pull request by szha**

### CODE REVIEW

File: python/mxnet/ndarray/ndarray.py [hunk 1]
```diff
@@ -673,6 +676,251 @@ def reshape(self, shape):
                                          ctypes.byref(handle)))
         return NDArray(handle=handle, writable=self.writable)
 
+    def flatten(self):
```
Comment (piiswrong) [inline python/mxnet/ndarray/ndarray.py]: you can simply do flatten = flatten

Comment (szha) [inline python/mxnet/ndarray/ndarray.py]: If I do this, do I still need to add doc string?

File: python/mxnet/ndarray/ndarray.py [hunk 2]
```diff
@@ -42,6 +42,9 @@
 from . import broadcast_sub, broadcast_div, broadcast_to, broadcast_equal, cast_storage
 from . import broadcast_greater, broadcast_greater_equal, broadcast_lesser, broadcast_lesser_equal
 from . import zeros_like, slice
+from . import flatten, expand_dims, tile, flip
```
Comment (piiswrong) [inline python/mxnet/ndarray/ndarray.py]: do you have to import these ops directly? How about import the namespace?

File: python/mxnet/ndarray/ndarray.py [hunk 3]
```diff
@@ -673,6 +671,26 @@ def reshape(self, shape):
                                          ctypes.byref(handle)))
         return NDArray(handle=handle, writable=self.writable)
 
+    flatten = flatten
```
Comment (piiswrong) [inline python/mxnet/ndarray/ndarray.py]: Does document display properly for these functions?

File: python/mxnet/ndarray/ndarray.py [hunk 4]
```diff
@@ -673,6 +686,100 @@ def reshape(self, shape):
...
+    def ceil(self, **kwargs):
+        """Convenience fluent method for ndarray.ceil.
+        See ceil"""
+        return op.ceil(self, **kwargs)
+
+    def trunc(self, **kwargs):
+        """Convenience fluent method for ndarray.trunc.
+        See trunc"""
+        return op.trunc(self, **kwargs)
+
```
Comment (reminisce) [inline python/mxnet/ndarray/ndarray.py]: Do we need to support `repeat` as well?

File: python/mxnet/ndarray/ndarray.py [hunk 5]
```diff
@@ -673,6 +686,100 @@ def reshape(self, shape):
                                          ctypes.byref(handle)))
         return NDArray(handle=handle, writable=self.writable)
 
+    def flatten(self, **kwargs):
```
Comment (reminisce) [inline python/mxnet/ndarray/ndarray.py]: Do we need to support positional arguments as well? At least, the following two usages of the op through `mx.nd` both work:
```python
mx.nd.tile(a, reps=(2, 3))
mx.nd.tile(a, (2, 3))
```

Comment (szha) [reply python/mxnet/ndarray/ndarray.py]: Will add. Thanks

File: python/mxnet/ndarray/ndarray.py (hunk 4 above)
Comment (szha) [reply python/mxnet/ndarray/ndarray.py]: I was limiting the methods to unary ops with 0 or 1 argument before. Let me revisit the op list and see what else can be added.

File: docs/api/python/ndarray.md [hunk 6]
```diff
@@ -118,7 +118,58 @@ We summarize the interface for each class in the following sections.
 
     NDArray.T
     NDArray.reshape
+    NDarray.flatten
+    NDarray.expand_dims
```
Comment (eric-haibin-lin) [inline docs/api/python/ndarray.md]: `NDArray` instead of `NDarray` :P

**Comment by szha** [comment]
Ready for review. Document now works and examples can be found at:
http://mxnet-doc.s3-website-us-east-1.amazonaws.com/api/python/ndarray.html#the-ndarray-class
http://mxnet-doc.s3-website-us-east-1.amazonaws.com/api/python/symbol.html#the-symbol-class

**Pull request merged**